superset/db_engine_specs.py

```diff
@@ -487,6 +488,16 @@ def get_timestamp_column(expression, column_name):
         can be overridden."""
         return expression or column_name
 
+    @classmethod
+    def make_validation_query(cls, sql):
+        """
+        If the underlying engine supports it, modify the query sql to request
+        that the database validate the query instead of running it.
+        """
+        raise Exception(
```

Comment:
I left this in as a defense in depth — right now it's impossible to hit these since we check the conditional in the query path before ever invoking this, but I figured it was either explicitly throw here or return something invalid that might get handed farther up, like a None, moving the 5xx trace farther from the source.